### ISSUE TITLE
Terraria second instance picking up input from the first one

### ISSUE THREAD

**Issue by Coacic** [UniversalSplitScreen/UniversalSplitScreen#7]
Hey I've tried your software on few games and it works like a charm, but for some reason I just couldn't get Terraria to work. I've set up two instances as usual set up recommended settings for it. Set for first instance first keyboard/mouse and for second instance I've set up second keyboard/mouse. And as it seems it works fine until I start clicking with my first mouse it starts screwing up and input starts to share between windows and I can control second character with first keyboard, even tho I'm not supposed to. And in menu i can control with first keyboard second window and go up or down. 

I'm using windows 10 and also installed pretty much every vc redists and latest .NET framework being 4.8.

**Comment by Ilyaki** [comment]
This seems to be caused by the GetKeyboardState function reading input from multiple keyboards. I've added a hook so it will be fixed in the next version. (You can compile it from the develop branch if you want to try it)

**Comment by Coacic** [comment]
Thanks for the reply bud,

I've tried to compile it but I'm not that great with that stuff, I assume I need visual studio with pre installed features and as it seemed It had installed correctly with vs tools v141 I believe, and I've tried to compile it by batch building, but it had returned many errors. I really don't know what to do, I'm clueless if you could shed me some light, would be helpful.

As for the bug, I'm glad it's getting fixed and thanks for being so awesome to make such an excellent software. 👍

**Comment by jerickski7** [comment]
I have an issue with terraria also but with gamepad , both instances picks up the input from a single gamepad that is setup for the first instance

**Comment by Ilyaki** [comment]
@Coacic 
Go to Tools -> NuGet Package Manager -> Manage NuGet packages for solution... and click 'Restore' somewhere at the top: that should download all the dependencies.

@jerickski7 
Do you have XInput hook enabled in options? And make sure for the first instance to set controller index 1, the second should be set to 2, etc.

**Comment by jerickski7** [comment]
Yes enabled Xinput hook , and set-up the gamepad 1 for the first instance and gamepad 2 on second instance

**Comment by Ilyaki** [comment]
@jerickski7 Try starting split screen after you have joined the server. Might be that xinput isn't loaded until after.

**Comment by Coacic** [comment]
@Ilyaki Thanks for the tips now it works. So far haven't found any bugs and it just works. Good job bro lovin it so far. 🔥 💯
